src/classes/CW_AudienceGroupControllerTest.cls

```diff
@@ -0,0 +1,333 @@
+*                  CIM, BI, Marketer must be able to View,Edit,Delete audience groups 
+*				   that are linked to the campaign
+*
+* Modification Log
+* ============================================================================
+* Ver Date       Author         Modification
+* --- ---------- -------------- --------------------------
+* 0.1 03/04/2014 Cristina P		Initial
+ */
+@isTest
+private class CW_AudienceGroupControllerTest 
+{
+
+	public static CW_Campaign_Navigation_Settings__c 	cns1;
+	public static CW_Campaign_Navigation_Settings__c 	cns2;
+	
+	public static CW_CampaignListSetting__c		audienceGroupPageSizeRecord;
+	
+	public static void initCustomSettings()
+	{
+		cns1 = TestDataUtil.createCampaignNavigationSettings(Page.CW_Campaign_Information,1,false);
+    	cns2 = TestDataUtil.createCampaignNavigationSettings(Page.CW_Audience,2,false);
+    	insert new List<CW_Campaign_Navigation_Settings__c>{cns1,cns2};
+    	
+    	audienceGroupPageSizeRecord = new CW_CampaignListSetting__c(Name=ConstantUtil.AUDIENCE_GROUP_PAGE_SIZE,
+    											 Integer_Value__c=10);
+    	insert audienceGroupPageSizeRecord;
+	}
+	
+	static testMethod void standardSetControllerTest() 
+	{
+        initCustomSettings();
+        
+        CW_Campaign__c camp = TestDataUtil.createCampaign();
+        insert camp;
+        
+        List<CW_Audience_Group__c> listGroups = TestDataUtil.createAudienceGroups(camp.Id,15);
+        insert listGroups;
+        
+        User thisUser = [SELECT Id FROM User WHERE Id = :UserInfo.getUserId()];
+    	
+    	Test.startTest();
+    	
+		System.runAs(thisUser)
+	    {  
+	    	audienceGroupPageSizeRecord.Integer_Value__c = 5;
+			update audienceGroupPageSizeRecord;
+			
+	    	Test.setCurrentPage(Page.CW_Campaign_Information);
+			CW_WizardController wizCtrl = new CW_WizardController(new ApexPages.StandardController(camp));
+			wizCtrl.next();
+			
+			Test.setCurrentPage(Page.CW_Audience);
+			Apexpages.currentPage().getParameters().put('id',camp.Id);
+			CW_AudienceGroupController agCtrl = new CW_AudienceGroupController();
+			
+			agCtrl.selectAll = true;
+			agCtrl.selectDeselectAllItems();
+			system.assertEquals(true,agCtrl.getHasNext());
+			
+			agCtrl.doNext();
+			
+			system.assertEquals(agCtrl.selectAll, false);
+			
+			system.assertEquals(true,agCtrl.getHasPrevious());
+			agCtrl.doPrevious();
+			
+			agCtrl.contextItem = agCtrl.campGroups[0].aGroup.Id;
+			agCtrl.doSelectItem();
+			agCtrl.doDeselectItem();
+			
+			agCtrl.last();
+			system.AssertEquals(agCtrl.setGroups.getRecords().size(),5);
+			
+			agCtrl.first();
+			system.AssertEquals(agCtrl.setGroups.getRecords().size(),5);
+			
+			agCtrl.doNext();
+			system.AssertEquals(agCtrl.hasFirstPage,true);
+			system.AssertEquals(agCtrl.hasLastPage,true);
+			
+			agCtrl.doPrevious();
+			system.AssertEquals(agCtrl.hasFirstPage,false);
+			system.AssertEquals(agCtrl.hasLastPage,true);
+			
+	    }
+	}
+	
+	static testMethod void editSaveGroupsTest() 
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility